File: apzkr-pzpi-21-3-dzabakhidze-davyd/Task1-Server/Kolosok.Domain/Entities/BrigadeRescuer.cs
using System.ComponentModel.DataAnnotations;

namespace Kolosok.Domain.Entities;

public class BrigadeRescuer : BaseEntity
{
    public Guid ContactId { get; set; }
    [Required]
    public Contact Contact { get; set; }
    
    public string Position { get; set; }
    public string Specialization { get; set; }
    
    public Brigade Brigade { get; set; }
    public Guid BrigadeId { get; set; }
    
    public ICollection<Action> Actions { get; set; }
    public ICollection<Victim> Victims { get; set; }
}

File: apzkr-pzpi-21-3-dzabakhidze-davyd/Task3-WebClient/src/app/models/brigade-rescuer.ts
import {Victim} from "./victim";
import {BaseEntity} from "./base-entity";
import {Contact} from "./contact";
import {Brigade} from "./brigade";
import {Action} from "./action";

export interface BrigadeRescuer extends BaseEntity {
  contactId: string;
  contact: Contact;
  position: string;
  specialization: string;
  brigadeId: string;
  brigade: Brigade;
  actions: Action[];
  victims: Victim[];
}
